### PR TITLE
Update `package.json` exports

### LINKED ISSUE #82: Problem with new `moduleResolution: bundler` (Typescript 5.0)
I noticed this error after adding `"moduleResolution": "bundler"` in my `tsconfig.json`.

solid-toast-module-resolution-bundler

I don't fully understand the error message, but I found an easy way to fix it by modifying `node_modules/solid-toast/package.json`. If I add this line, the error instantly goes away:

```diff
  "exports": {
    ".": {
+      "types": "./dist/types/index.d.ts",
      "solid": "./dist/source/index.jsx",
      "import": "./dist/esm/index.js",
      "browser": "./dist/esm/index.js",
      "require": "./dist/cjs/index.js",
      "node": "./dist/cjs/index.js"
    }
  },
```

I'll put it in a PR if this seems like the proper fix.

### PR DESCRIPTION
**Pull request by nathanbabcock**
@ardeora let me know if you'd like any changes to these exports (or feel free to modify the branch directly). I adapted them from `solid-lib-starter` as you recommended. I tried the new `package.json` locally and it fixes #82.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,13 +10,22 @@
...
+        "types": "./dist/types/index.d.ts",
+        "default": "./dist/esm/index.js"
+      },
+      "require": "./dist/cjs/index.js"
+    },
+    "import": {
+      "types": "./dist/types/index.d.ts",
+      "default": "./dist/esm/index.js"
+    },
+    "require": "./dist/index.cjs"
```
Comment (ardeora) [inline package.json:28]:
```suggestion
    "require": "./dist/cjs/index.js"
```

Thanks for working on this! Just a small change in the `require` export condition. Also I have published `0.5.0-alpha.0` with your changes. Could you confirm if the alpha package solve your issue as well?

**Review by ardeora — CHANGES_REQUESTED** [review]

**Comment by ardeora** [comment]
Oh missed that I could edit the PR. I just committed the fix. Thanks again!

**Comment by nathanbabcock** [comment]
Np, thanks for the sweet library 😄

**Comment by ardeora** [comment]
Released `v0.5.0`! 🚀

**Pull request merged**